### DISCUSSION TITLE
Replying emails

### DISCUSSION THREAD

**Discussion by Cyb3r-Monk** [cmu-sei/GHOSTS#109] [Q&A] [answered]
I'm using the Outlook timeline config shared here: https://github.com/cmu-sei/GHOSTS/blob/d31f6f45f146d1b4a3a302e55881d26cdfce7772/src/Ghosts.Client/Sample%20Timelines/Outlook.json

Sending emails inside the domain works but when sending an email from outside domain, the email doesn't get replied. Also, No emails are sent to the outside domain. How should I configure email reply and outside email send&reply? I couldn't find information in the Wiki.

**Comment by sei-dupdyke** [comment]
Pls standby, this will take me a bit to setup and debug.

**Reply by Cyb3r-Monk** [reply]
Btw, I downloaded the latest versions of Redempiton DLLs, and the lib folder items are:
Interop.Redemption.dll (downloaded from the Redemption website)
Redemption.dll (downloaded from the Redemption website)
Redemption64.dll (downloaded from the Redemption website)

Office version: 
Microsoft® Outlook® for Microsoft 365 MSO (Version 2207 Build 16.0.15427.20182) 64-bit 

Should I use the DLL files that come within the client?

**Reply by sei-dupdyke** [reply]
Hmm, not sure about 365, I have not tested with that distribution 🤔

(But that seems like it shouldn't affect some things working and others not working — instead it would be that nothing worked at all.)

**Reply by sei-dupdyke** [reply]
Quick fix attempt here: You must use the x64 version of GHOSTS in order to use x64 office (bitness must match).

**Comment by sei-dupdyke** [comment]
There are a couple of settings to review in config/application.json:

"RecipientsOutsideMin": 0,
"RecipientsOutsideMax": 1,

Change the min value to 1 and see if that corrects things please.

**Comment by sei-dupdyke** [comment, marked as answer]
Aha, I found the bug and fixed it in patch 6.1.8.0. Sorry for the confusion.

**Reply by Cyb3r-Monk** [reply]
I'm observing a strange behaviour:
I changed the "RecipientsOutsideMin" from 0 to 1. Agent started sending emails outside. However, it started sending emails every 60minutes although I have the delay configured to 30minutes. 
Apart from that, replying emails is still not working. I'm using the x64 binary.

**Reply by sei-dupdyke** [reply]
Can I please see your timeline for this Outlook handler?

**Reply by Cyb3r-Monk** [reply]
Sure! 
In the email reply code, reply section is written uppercase(case 'REPLY': ). I thought that might be an issue and used uppercase as well. 

```JSON
    {
     "HandlerType": "Outlook",
     "Initial": "",
     "UtcTimeOn": "08:00:00",
     "UtcTimeOff": "21:30:00",
     "Loop": "True",
     "TimeLineEvents": [
         {
             "Command": "create",
             "CommandArgs": [
                 "CurrentUser",
                 "random",
                 "random",
                 "random",
                 "Random",
                 "Random",
                 "PlainText",
                 ""
             ],
             "DelayAfter": 1800000,
             "DelayBefore": 300000
         },
         {
             "Command": "REPLY",
             "CommandArgs": [
                 "CurrentUser",
                 "All",
                 "All",
                 "All",
                 "Parent",
                 "Parent+Random",
                 "Parent",
                 ""
             ],
             "DelayAfter": 1800000,
             "DelayBefore": 240000
         }
     ]
    }
```

**Reply by Cyb3r-Monk** [reply]
6.1.11 seems to fix half of the replying issue. Now I have a strange situation:

I sent an email to the agent from outside domain. On the agent machine:
I have one email draft in the Sent Items folder and another email (the same as draft) which was sent. The original received email is still unread in the inbox.  Also the reply text is in the wrong place. Is the above reply config OK? What do all arguments mean?

<img width="630" alt="Screenshot 2022-08-25 at 08 24 54" src="https://user-images.githubusercontent.com/26627224/186591131-068cfd1b-5a92-42b9-8b16-d06adbcec147.png">

**Reply by sei-dupdyke** [reply]
This is fixed in 6.1.12. All this reminds me the outlook code hasn't been touched since the beginning of time.

**Reply by Cyb3r-Monk** [reply]
Works like a charm! Thanks a lot, Dustin!

**Reply by Cyb3r-Monk** [reply]
Hmm. Seems like the normal send email function stopped putting any content except the confidential signature into the emails. 

<img width="703" alt="image" src="https://user-images.githubusercontent.com/26627224/186925580-492c1280-2d2b-4104-84ff-6ff361574601.png">

**Reply by sei-dupdyke** [reply]
This is fixed in 6.1.15. You may still get reply content in a "create" email, but this is a known issue that I can't prioritize at the moment.

**Reply by Cyb3r-Monk** [reply]
Thanks! Works way better now.
